### PR TITLE
Building manual

### PR DESCRIPTION
**Pull request by tomsmulski**
here are the commits of building_navigation. don't get confused

The building manuel is done xD

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -108,12 +111,21 @@ export default function App() {
 
   if (!loadingStatus.status) {
     return (
-      <>
+      <StyledMain>
         <ResourcesOverview />
-        <Building />
-      </>
+        <span>
```
Comment (actyralf) [inline src/App.js]: This `span` does not seem to be necessary... Can it be removed?

File: src/App.js [hunk 2]
```diff
@@ -108,12 +111,21 @@ export default function App() {
...
+          <Building selectedBuilding={selectedBuilding} />
+          <Buildingnav currentUserBuildings={currentUserBuildings} selectedBuilding={selectedBuilding} />
+        </span>
+      </StyledMain>
     );
   } else {
     return <Loading></Loading>;
```
Comment (actyralf) [inline src/App.js]: Can be self closing: `<Loading/>`

File: src/GlobalStyles.js [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 // Trick prettier into formatting "createGlobalStyle"
 const styled = {createGlobalStyle};
 
 export default styled.createGlobalStyle`
   * {
     box-sizing: border-box;
-    margin: 0;
     padding: 0;
+    max-width: 800px;
+    margin: 0 auto;
```
Comment (actyralf) [inline src/GlobalStyles.js:10]: Do you really want **every element** to have `margin: 0 auto` and `max-width: 800px`?
It would be better to use the `*` selector for a css reset (e.g. `padding:0; margin: 0;`) and to do the specific styling where needed.

File: src/components/building/building.js [hunk 4]
```diff
@@ -40,78 +41,57 @@ export const Building = () => {
...
+        buildType={0}
+        buildId={gameBuildingType.id}
+        buildName={gameBuildingType.name}
+        buildDescription={gameBuildingType.description}
+        buildTime={buildTime}
+        buildPrice={gameBuildingType.buildMaterials}
+        buildPriceTextColor={checkEnoughRescource.buildPriceTextColor}
+        buildYield={gameBuildingType.productionMaterials}
+        currentBuildLevel={currentUserBuilding.level}
+        buttonText={buttonTexts}
```
Comment (actyralf) [inline src/components/building/building.js]: `buttonTexts` is a string and not an array, right? Then it would be better to name it in singular: `buttonText`.
Maybe it would be cleaner to pass only a state like `inProgress` to the `BuildingCard` component and let the component handle the Button state and text.

File: src/components/building/building.js [hunk 5]
```diff
@@ -40,78 +41,57 @@ export const Building = () => {
...
-          currentBuildLevel={currentUserBuilding.level}
-          buttonText={buttonTexts}
-          buttonDisabled={notEnoughRescourceButtonDisable || buildInProgressButtonDisable }
-          buttonFunction={onHandleClickUpgrade}
-          currentUserBuildings={currentUserBuildings}
-        />
-      </article>
-    );
+  if (currentUserBuildingInProgress.length > 0) {
+    const inProgressId = currentUserBuildingInProgress[0].buildingId;
```
Comment (actyralf) [inline src/components/building/building.js:53]: If one building is "in progress", it's not possible to build another building simultaneously. Is that the desired flow?

File: src/components/building/buildingcard.js [hunk 6]
```diff
@@ -13,100 +18,111 @@ export default function Buildingcard({
   buildPrice,
   buildPriceTextColor,
   buildYield,
-  buildInProgressTime,
   currentBuildLevel,
   buttonText,
   buttonDisabled,
   buttonFunction,
```
Comment (actyralf) [inline src/components/building/buildingcard.js]: Instead of `buttonFunction` you should use an event-like name, e.g. `onActionButtonClick`

File: src/components/building/buildingcard.js [hunk 7]
```diff
@@ -120,16 +136,18 @@ const StyledBuildInfoContainer = styled.article`
...
   display: flex;
   flex-direction: column;
-  padding: 5px;
+  min-height: 158px;
+  margin: 0;
 `;
 
 const StyledSpan = styled.span`
   display: flex;
   justify-content: space-between;
```
Comment (actyralf) [inline src/components/building/buildingcard.js:146]: You shouldn't use a `span` as a `flex` container as you usually expect a span to have `display: inline`

File: src/components/button/button.js [hunk 8]
```diff
@@ -2,7 +2,13 @@ import styled from 'styled-components';
 
 export default function Button({disabled, buildId, buildTime, onHandleClick, buttonText}) {
   return (
-    <StyledButton type='button' disabled={disabled} data-buildid={buildId} data-buildtime={buildTime} onClick={onHandleClick}>
+    <StyledButton
+      type="button"
+      disabled={disabled}
+      data-buildid={buildId}
+      data-buildtime={buildTime}
```
Comment (actyralf) [inline src/components/button/button.js]: What is the purpose of the `data-...` attributes? It should not be necessary to store data in DOM elements like this.

**Review by actyralf — APPROVED** [review]
This is a **huge** pull request, so it's hard to follow all changes you did.
Regarding styling: You use a lot of individual styled components - maybe you can use less and configure them via props? For general components like `StyledH2` etc. it could make sense to either style these in GlobalStyles or have separate component files for them so that they can be reused from all components.

File: src/App.js (hunk 1 above)
Comment (tomsmulski) [reply src/App.js]: yes the is useless there, is removed

File: src/App.js (hunk 2 above)
Comment (tomsmulski) [reply src/App.js]: change it

File: src/components/building/building.js (hunk 5 above)
Comment (tomsmulski) [reply src/components/building/building.js:53]: I think because of index 0. that is already pre-built if you have several cities

File: src/components/building/buildingcard.js (hunk 6 above)
Comment (tomsmulski) [reply src/components/building/buildingcard.js]: done. buttonFunction to onActionButtonClick

File: src/components/button/button.js (hunk 8 above)
Comment (tomsmulski) [reply src/components/button/button.js]: I understand this now
you mean to say that I should remove the data-... and then insert it directly into the onActionButtonClick ?

I pass only the parameters. but yes now that I write this. sounds logical

File: src/components/building/buildingcard.js (hunk 7 above)
Comment (tomsmulski) [reply src/components/building/buildingcard.js:146]: with inline, it doesn't look like it should. but since it's part of the style, i'll postpone the whole thing until thursday, when i'll really start with style.

image

File: src/GlobalStyles.js (hunk 3 above)
Comment (tomsmulski) [reply src/GlobalStyles.js:10]: yes i did the whole thing for now so that it looks like anything at all. without that everything is left side bound. from thursday i'll implement the styling

File: src/components/button/button.js (hunk 8 above)
Comment (tomsmulski) [reply src/components/button/button.js]: thanks, done

File: src/components/building/building.js (hunk 4 above)
Comment (tomsmulski) [reply src/components/building/building.js]: that with the button text, there I just on the fast no suitable simple solution, there I would lose with thinking too much time. I would like to leave that so for now as long as it works so.

as soon as there is some free space i will adjust it, because more than half of the code can be outsourced anyway.

File: src/components/building/buildingcard.js (hunk 7 above)
Comment (actyralf) [reply src/components/building/buildingcard.js:146]: I think you got me wrong: You should use `flex` and `space-between` but I think it's more common to use a `div` instead of a `span` for that.

**Review by actyralf — APPROVED** [review]
